src/components/molecules/AnnouncementMessage/AnnouncementMessage.tsx

```diff
@@ -1,51 +1,89 @@
-import React, { useCallback, useEffect, useState } from "react";
-import classNames from "classnames";
+import React, { useEffect } from "react";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";
+import classNames from "classnames";
 
 import { getLinkFromText } from "utils/getLinkFromText";
+import { isDefined } from "utils/types";
+
+import { useShowHide } from "hooks/useShowHide";
 
-import { useChatSidebarControls } from "hooks/chatSidebar";
+import { Banner } from "types/banner";
+
+import { LinkButton } from "components/atoms/LinkButton";
 
 import "./AnnouncementMessage.scss";
 
-type AnnouncementMessageProps = {
-  message?: string;
-};
+export interface AnnouncementMessageProps {
+  banner?: Banner;
+  announcementForUser?: boolean;
+}
 
 export const AnnouncementMessage: React.FC<AnnouncementMessageProps> = ({
-  message = "",
+  banner,
+  announcementForUser = false,
 }) => {
-  const [isVisible, setVisibility] = useState<boolean>(false);
-  const { isExpanded } = useChatSidebarControls();
-
-  const hideAnnouncement = useCallback(() => {
-    setVisibility(false);
-  }, []);
+  const {
+    isShown: isShownAnnouncementMessage,
+    show: showAnnouncementMessage,
+    hide: hideAnnouncementMessage,
+  } = useShowHide();
 
   useEffect(() => {
-    if (message) {
-      setVisibility(true);
+    if (isDefined(banner?.content)) {
+      showAnnouncementMessage();
     }
-  }, [message]);
+  }, [banner, showAnnouncementMessage]);
 
-  if (!isVisible || !message) return null;
+  const noop = (e: React.MouseEvent<HTMLElement>) => {
+    e.stopPropagation();
+  };
```

Comment:
What's the intent/purpose of the `noop` function? It probably shouldn't be defined inline within this component. If it's needed it should just be something like `const noopFunc = () => {}` near the top of the file.

Why `e.stopPropagation();` here? (cc // @sunny-viktoryia @mike-lvov: Pretty sure this has been a pattern we've been trying to get rid of?)

Reply:
yes, we try to avoid using `e.stopPropagation()` in favor of `e.closest()`
(related discussion: https://github.com/sparkletown/sparkle/pull/1302#discussion_r628998528)

instead of `e.stopPropagation()` is it possible to use `e.closest()` in the place where we want to prevent a function to be executed?

Reply:
removed in 9937587